### PR TITLE
Update Person Card

### PR DESCRIPTION
**Pull request by ziyang27**
The few changes I made:
- Remove the address and email from the person's card
- Make the visibility of the upcoming and last-contact dynamic

### CODE REVIEW

**Comment by ziyang27** [comment]
Note: The patch coverage will not be high. I touched the Ui components. However, the code coverage is expected to increase.

**Review by tanyunchao — APPROVED** [review]
Good work on this brother, LGTM!

File: src/main/java/seedu/address/model/person/Person.java [hunk 1]
```diff
@@ -72,10 +72,18 @@ public Upcoming getUpcoming() {
         return upcoming;
     }
 
+    public boolean hasUpcoming() {
+        return upcoming.hasUpcoming();
+    }
+
```
Comment (tanyunchao) [inline src/main/java/seedu/address/model/person/Person.java:78]: thanks for helping to add this

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 2]
```diff
@@ -50,8 +50,7 @@ public AddCommand parse(String args) throws ParseException {
         Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
         Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
         Upcoming upcoming = ParserUtil.parseUpcoming(argMultimap.getValue(PREFIX_UPCOMING).orElse(""));
-        LastContact lastcontact = ParserUtil.parseLastContact(argMultimap.getValue(PREFIX_LASTCONTACT)
-                .orElse(""));
+        LastContact lastcontact = ParserUtil.parseLastContact(argMultimap.getValue(PREFIX_LASTCONTACT).orElse(""));
```
Comment (SimHanKiong) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:53]: Nice QOL change

**Review by SimHanKiong — COMMENTED** [review]
Nice changing the UI, really good job

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java (hunk 2 above)
Comment (tanyunchao) [reply src/main/java/seedu/address/logic/parser/AddCommandParser.java:53]: i think @SimHanKiong 's point here cannot be understated.

Comment (Colex2000) [reply src/main/java/seedu/address/logic/parser/AddCommandParser.java:53]: Thanks bro

File: src/main/java/seedu/address/ui/PersonCard.java
Comment (Colex2000) [inline src/main/java/seedu/address/ui/PersonCard.java:1]: Seems like u managed to make the upcoming and last contact field disappear if theres no datetime to it. Nice work bro

**Review by Colex2000 — COMMENTED** [review]
GOOD WORK

**Pull request merged**